src/config/watcher.rs

```
@@ -269,8 +268,23 @@ mod tests {
         {
             panic!("Test timed out");
         }
+    }
+    #[tokio::test]
```

Thank you for adding a test and fixing the failure. I realize that these tests have several problems (existing, not introduced by this PR): 

* they are disabled on macOS
* they have arbitrary delays e.g. 3 seconds and then 3*5 
* they panic instead of returning an error

But again, this is out of scope for this PR.